### PR TITLE
Config

### PR DESCRIPTION
**Pull request by sadrasabouri**
#### Reference Issues/PRs
#43 
#### What does this implement/fix? Explain your changes.
The process for saving and loading a config file is now such that you can use either `--config` argument while running `pyrgg` or choose your desired config (which should be located at the source where you are running `pyrgg`). You will be asked as the final question if you wanted to save the config by which you generate graph(s) each time after generation. Below changes are also applied:
- `check_for_config` function
- `load_config` function
- `save_config` function

#### Any other comments?
The only point I missed is the test for `_print_select_config` in the case where' configs' are not empty. I tried the below pytest code but it failed, I don't know why. Maybe @sepandhaghighi can help me with this. BTW I went with `# pragma: no cover` for now.
```python
>>> loaded_config = check_for_config(input_func_conf_test)
Config files detected in current directory are listed below:
[0] - (.*)
Press the config index to load or any other keys to start with new one : 
```

### CODE REVIEW

**Comment by sadrasabouri** [comment]
> @sadrasabouri Thanks for your efforts rocket
> 
>     1. About the test issue:
>        
>        1. Add `ELLIPSIS` to the `doctest_optionflags` section in `pytest.ini`
>        2. Use this format : `[0] - ...`
Not working :confused:
I'm still getting errors.
>        3. Add a new function like `def input_func_conf_test2(input_data)` to always return 1
> 
>     2. Add `pyrgg_version` to the config file
> 
Working on it.
>     3. I'm not okay with the `output_format` field in the config file, it should be more transparent, something like : `"output_format": "JSON"`
Working on it.

**Comment by sepandhaghighi** [comment]
@sadrasabouri Take a look at the outputs :

```
Expected:
    Config files detected in current directory are listed below:
    [0] - ...
    Press the config index to load or any other keys to start with new one : 
Got:
    Config files detected in current directory are listed below:
    [0] - /home/runner/work/pyrgg/pyrgg/File 1.pyrgg.config.json
 ```

it seems you should remove this line `Press the config index to load or any other keys to start with new one : `

File: pyrgg/params.py [hunk 1]
```diff
@@ -62,13 +63,20 @@
     16: ".gv"
 }
 
+OUTPUT_FORMAT = {i: output_format[1:].upper()
+                 for i, output_format in SUFFIX_MENU.items()}
+
+OUTPUT_FORMAT_inv = {v: k for k, v in OUTPUT_FORMAT.items()}
```
Comment (sepandhaghighi) [inline pyrgg/params.py:69]: `OUTPUT_FORMAT_INV`

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@sadrasabouri Take a look at my comment about constant name 🔥

File: pyrgg/__main__.py [hunk 2]
```diff
@@ -87,15 +87,21 @@ def gen_graph(input_dict, file_name):
...
+    :type input_dict: dict
     :return: None
     """
-    input_dict = get_input()
+    if input_dict is None:
+        input_dict = get_input()
     file_name = input_dict["file_name"]
     number_of_files = input_dict["number_of_files"]
+    if input_dict["config"] is True:
+        print("Config --> " + save_config(input_dict))
```
Comment (sepandhaghighi) [inline pyrgg/__main__.py:103]: @sadrasabouri I want to merge this PR, but before that I have a question:
Is it okay to re-save the config file? (when load data from config)
It's a bit weird 😕

Comment (sadrasabouri) [reply pyrgg/__main__.py:103]: This way the code architecture remains unified. I believe there is no problem (except facing an error while saving it).

If you are thinking of any better solution please let me know.

File: pyrgg/__main__.py [hunk 3]
```diff
@@ -87,15 +87,21 @@ def gen_graph(input_dict, file_name):
...
     """
     Run proper converter.
 
+    :param input_dict: input data
+    :type input_dict: dict
     :return: None
     """
-    input_dict = get_input()
+    if input_dict is None:
+        input_dict = get_input()
```
Comment (sepandhaghighi) [inline pyrgg/__main__.py:99]: Add `load_from_config = False`

File: pyrgg/__main__.py [hunk 4]
```diff
@@ -87,15 +87,21 @@ def gen_graph(input_dict, file_name):
...
     file_name = input_dict["file_name"]
     number_of_files = input_dict["number_of_files"]
+    if input_dict["config"] is True:
```
Comment (sepandhaghighi) [inline pyrgg/__main__.py:102]:
```python
if input_dict["config"] is True and not load_from_config
```

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]

**Review by sepandhaghighi — APPROVED** [review]

**Pull request merged**